populator-machinery/controller.go

```diff
@@ -527,13 +556,49 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 	}
 
 	// Look for the populator pod
-	podName := fmt.Sprintf("%s-%s", populatorPodPrefix, pvc.UID)
-	c.addNotification(key, "pod", c.populatorNamespace, podName)
 	var pod *corev1.Pod
-	pod, err = c.podLister.Pods(c.populatorNamespace).Get(podName)
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			return err
+	podName := fmt.Sprintf("%s-%s", populatorPodPrefix, pvc.UID)
+	if c.usePod {
+		c.addNotification(key, "pod", c.populatorNamespace, podName)
+		pod, err = c.podLister.Pods(c.populatorNamespace).Get(podName)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return err
+			}
+		}
+	}
+
+	// If populate data without populator pod, and orginal StorageClass's VolumeBindingMode is VolumeBindingWaitForFirstConsumer,
+	// create a StorageClass with VolumeBindingImmediate for pvcPrime
+	storageClassPrimeName := *pvc.Spec.StorageClassName
+	if !c.usePod && storageClass.VolumeBindingMode != nil && storagev1.VolumeBindingWaitForFirstConsumer == *storageClass.VolumeBindingMode {
+		storageClassPrimeName = populatorStorageClassPrefix + "-" + *pvc.Spec.StorageClassName
+		scPrime, err := c.scLister.Get(storageClassPrimeName)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return err
+			}
+			c.addNotification(key, "sc", "", storageClassPrimeName)
+			bm := storagev1.VolumeBindingImmediate
+			scPrime = &storagev1.StorageClass{
+				ObjectMeta:           metav1.ObjectMeta{Name: storageClassPrimeName},
```

Comment:
we should also add an annotation to clearly identify the the volume populator is the owner of this SC (and not confuse with any user create SC which has the same prefix)

Reply:
See https://github.com/kubernetes-csi/lib-volume-populator/pull/133#discussion_r1501108362